src/NATS.Client.JetStream/Models/StreamSource.cs

```diff
@@ -54,4 +54,7 @@ public record StreamSource
     [System.Text.Json.Serialization.JsonPropertyName("external")]
     [System.Text.Json.Serialization.JsonIgnore(Condition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingDefault)]
     public ExternalStreamSource? External { get; set; }
+
+    [System.Text.Json.Serialization.JsonIgnore]
+    public string? Domain { get; set; }
```

Comment:
why do we need this? sorry not too familiar with mirroring

Reply:
I took it from the go implementation here https://github.com/nats-io/nats.go/blob/a06b6a92ee7d6c70e7348bcfdb719691ba1b4b63/jsm.go#L276

I believe it's used in Jetstream domains to source data from a different domain.  E.g. creating a stream in a leaf node that's sourced from say the main cluster e.g. NGS.  

Was surprised it wasn't serialized though....

Reply:
thanks for checking other implementations. my preference would be not to pollute the json model if possible. btw I checked [js](https://github.com/nats-io/nats.deno/blob/72cd3e52626b1a239d3b45c2e159cec513c7a477/jetstream/jsapi_types.ts#L324-L330) and [rust](https://github.com/nats-io/nats.rs/blob/ea4c372ba71a5528b9c24bb122a8806cf26d9187/async-nats/src/jetstream/stream.rs#L1551) they seem to have it too 😕 cc @aricart @Jarema

Reply:
we can keep it same as other clients for consistency.

Reply:
I've just started chartering some of these waters. Tossing some tidbits out at an 80% confidence level.

If the server + node are using domains, then the typical `$JS.API` isn't used when sending traffic to the leaf. Thus, in order to mirror the stream, you need the prefix in order to receive the the `$JS` traffic.

It shouldn't need to be sent to the server (or leaf in my example) since it is already configured to use the prefix in it's configuration.

Reply:
I've just started chartering some of these waters. Tossing some tidbits out at an 80% confidence level.

If the server + node are using domains, then the typical `$JS.API` isn't used when sending traffic to the leaf. Thus, in order to mirror the stream, you need the prefix in order to receive the the `$JS` traffic.

It shouldn't need to be sent to the server (or leaf in my example) since it is already configured to use the prefix in it's configuration.

Reply:
correct. 'domain' field is not needed or used by the server.